plugins/linux-spi-lpc/fu-plugin-linux-spi-lpc.c

```diff
@@ -0,0 +1,122 @@
+/*
+ * Copyright (C) 2020 Richard Hughes <richard@hughsie.com>
+ *
+ * SPDX-License-Identifier: LGPL-2.1+
+ */
+
+#include "config.h"
+
+
+#include "fu-plugin-vfuncs.h"
+#include "fu-hash.h"
+
+#define FU_PLUGIN_LINUX_SPI_LPC_SYSFS_DIR	"/sys/kernel/security/spi"
+
+void
+fu_plugin_init (FuPlugin *plugin)
+{
+	fu_plugin_set_build_hash (plugin, FU_BUILD_HASH);
+	if (fu_common_is_cpu_intel ())
+		fu_plugin_add_rule (plugin, FU_PLUGIN_RULE_REQUIRED_FOR, "hsi");
+}
+
+gboolean
+fu_plugin_startup (FuPlugin *plugin, GError **error)
+{
+	if (!g_file_test (FU_PLUGIN_LINUX_SPI_LPC_SYSFS_DIR, G_FILE_TEST_IS_DIR)) {
+		g_set_error (error,
+			     G_IO_ERROR,
+			     G_IO_ERROR_NOT_FOUND,
+			     "Cannot find %s",
+			     FU_PLUGIN_LINUX_SPI_LPC_SYSFS_DIR);
+		return FALSE;
+	}
```

Comment:
Checking this on a real system without the kernel patches this actually causes a runtime failure which I think is wrong, no?  Should you actually be able to get HSI1+UA! with a missing kernel module to really calculate or should this be HSI0+UA!?

Reply:
I think `HSI1+UA!` is expected when the kernel module is missing. The `!` kinda means "we don't trust this value" rather than "we failed one of the HSI:1 tests" -- logically we couldn't run the test, rather that it failing.

Reply:
OK that makes more sense.  I think that we should embellish this detail more clearly in the spec too then.